## 📌 Data Collection

In this step, historical price and dividend data for selected ETFs are collected using Python.

## 🛠️ Libraries Used

- pandas: Data manipulation and analysis  
- yfinance: Financial data retrieval  
- matplotlib: Data visualization  

In [1]:
import yfinance as yf
import pandas as pd

etfs = ["SCHD", "QQQ", "JEPQ"]

price_data = {}
div_data = {}

for etf in etfs:
    ticker = yf.Ticker(etf)
    
    df = ticker.history(start="2020-01-01")
    if df.empty:
        continue

    price_data[etf] = df["Close"]

    div = ticker.dividends
    div_data[etf] = div

price_df = pd.DataFrame(price_data)

print(price_df.shape)
print(price_df.head())
print(price_df.isnull().sum())

(1562, 3)
                                SCHD         QQQ  JEPQ
Date                                                  
2020-01-02 00:00:00-05:00  15.622281  208.580231   NaN
2020-01-03 00:00:00-05:00  15.509286  206.669632   NaN
2020-01-06 00:00:00-05:00  15.506599  208.001312   NaN
2020-01-07 00:00:00-05:00  15.485080  207.972336   NaN
2020-01-08 00:00:00-05:00  15.544265  209.535507   NaN
SCHD      0
QQQ       0
JEPQ    589
dtype: int64


## 🔍 Data Check Result

- Total rows: 1562  
- No missing values in SCHD and QQQ  
- JEPQ has 589 missing values due to its later inception date (launched in 2022) 
- Missing values in JEPQ are expected since it was listed after 2020

## 📊 Data Description

- Price data is collected using the history() function from yfinance  
- Closing prices are used to represent daily ETF values  
- Dividend data is collected separately using the dividends attribute  
- The data is stored in pandas DataFrame format for further analysis  